docs/tutorials/Remark.adoc

```diff
@@ -0,0 +1,396 @@
+
+    public static final String COMMAND_WORD = "remark";
+
+    @Override
+    public CommandResult execute(Model model) {
+        return new CommandResult("Hello from remark");
+    }
+}
+----
+
+=== Hook `RemarkCommand` into the application
+
+Now that we have our `RemarkCommand` ready to be executed, we need to update `AddressBookParser#parseCommand()` to recognize the `remark` keyword.
+Add the new command to the `switch` block by creating a new `case` that returns a new instance of `RemarkCommand`.
+
+You can refer to the changes in this link:https://github.com/j-lum/addressbook-level35/commit/72256eacea79dfcf077f5c7cb6f89819c02f090e#diff-5338391f3f6fbb4022c44add6590b74f[diff].
+
+=== Run the application
+
+Run `Main#main` and try out your new `RemarkCommand`.
+If everything went well, you should see something like this:
+
+.Output displayed
+image::RemarkHello.png[]
+
+== Change `RemarkCommand` to throw an exception
+
+While we have successfully printed a message to `ResultDisplay`, the command does not do what it is supposed to do.
+Let's change the command to throw an `CommandException` to accurately reflect that our command is still a work in progress.
+
+.The relationship between RemarkCommand and Command
+image::CommandInterface.png[]
+
+Following the convention in other commands, we add relevant messages as constants and use them.
+
+.RemarkCommand.java
+[source, java]
+----
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the remark of the person identified "
+            + "by the index number used in the last person listing. "
+            + "Existing remark will be overwritten by the input.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "r/ [REMARK]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + "r/ Likes to swim.";
+
+    public static final String MESSAGE_NOT_IMPLEMENTED_YET = "Remark command not implemented yet";
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        throw new CommandException(MESSAGE_NOT_IMPLEMENTED_YET);
+    }
+----
+
+== Enhancing `RemarkCommand`
+
+Let's change `RemarkCommand` to  parse input from the user.
+
+=== Make the command accept parameters
+
+We start by modifying the constructor of `RemarkCommand` to accept an `Index` and a `String`.
+While we are at it, let's change the error message to echo the values.
+While this is not a replacement for tests, it is an obvious way to tell if our code is functioning as intended.
+
+[source, java]
+----
+import static java.util.Objects.requireNonNull;
+//...
+import seedu.address.commons.core.index.Index;
+// ...
+private final Index index;
+private final String remark;
+
+/**
+ * @param index of the person in the filtered person list to edit the remark
+ * @param remark of the person to be updated to
+ */
+public RemarkCommand(Index index, String remark) {
+    requireAllNonNull(index, remark);
```

Comment:
Is not consistent with line 113.

I used `requireNonNull` until I realised later on at the last sections of the exercise to import requireAllNonNull from `seedu.address.commons.util.CollectionUtil.requireAllNonNull;`